### PR TITLE
Link source to targets via intermediate items in item-matrix

### LINKED ISSUE #199: Link targets via intermediate items in item-matrix
When this is available any table can be presented with a given set of intermediate item and (defined) relations.

### PR DESCRIPTION
**Pull request by JasperCraeghs**
Resolves #199

On top of #201

### CODE REVIEW

**Comment by JasperCraeghs** [comment]
Do we want to support external relationships between the intermediate and target? I haven't implemented this (yet). @Letme

**Comment by Letme** [comment]
I would say yes, but it can be implemented in the next pr to keep this a bit cleaner and smaller

File: mlx/directives/item_matrix_directive.py [hunk 1]
```diff
@@ -57,14 +57,16 @@ def perform_replacement(self, app, collection):
...
         # - No target filtering is done on external references
-
-        relationships = self['type']
-        if not relationships:
+        mapping_via_intermediate = {}
+        if not self['type']:
             # if no explicit relationships were given, we consider all of them (except for external ones)
             relationships = [rel for rel in collection.iter_relations() if not self.is_relation_external(rel)]
-            external_relationships = []
+            ext_relationships = []
```
Comment (Letme) [inline mlx/directives/item_matrix_directive.py:64]: I think external relationship name we had before is more descriptive...

File: doc/integration_test_report.rst [hunk 2]
```diff
@@ -343,6 +343,17 @@ Traceability from integration tests to requirements
     :sourcetype: impacts_on
     :stats:
 
+Traceability via intermediate items
+-----------------------------------
+
+.. item-matrix::
+    :source: DESIGN-
+    :target: UTEST ITEST
```
Comment (Letme) [inline doc/integration_test_report.rst:351]: You are missing intermediate flag here

Comment (JasperCraeghs) [reply doc/integration_test_report.rst:351]: I've made it optional. The source and target options are optional as well.

Comment (Letme) [reply doc/integration_test_report.rst:351]: Its example of the intermediate matrix so i would assume we want that demonstrated here...

Comment (JasperCraeghs) [reply doc/integration_test_report.rst:351]: Yes, I do. I'll add it. 

The problem with this example is that the DESIGN items are linked to mostly top-level RQT items with child RQT items. Those child RQT items are validated by TEST items, but not the top-level RQT items. This results in a rather empty item-matrix. Is this an issue that can and should be addressed?

Comment (JasperCraeghs) [reply doc/integration_test_report.rst:352]: Added intermediate flag in 4bc55e5c68e232847e16b67bc555603102f362b5.

File: mlx/directives/item_matrix_directive.py (hunk 1 above)
Comment (JasperCraeghs) [reply mlx/directives/item_matrix_directive.py]: I agree. Done in cb2a31a9e4beb210e9245dec414b11be1ce2f334.

File: mlx/directives/item_matrix_directive.py [hunk 3]
```diff
@@ -131,6 +125,149 @@ def perform_replacement(self, app, collection):
...
+            group (str): Optional group option
+        """
+        if not group:
+            tbody += rows.sorted
+        elif group == 'top':
+            tbody += rows.uncovered
+            tbody += rows.covered
+        elif group == 'bottom':
+            tbody += rows.covered
+            tbody += rows.uncovered
```
Comment (Letme) [inline mlx/directives/item_matrix_directive.py:144]: Wouldnt a return tbody here be better option? Then you don't need to pass in body, but you `+=` it on return part...

Comment (Letme) [reply mlx/directives/item_matrix_directive.py:144]: This is a general question as I see below we actually have a couple of other functions with the same pattern.

Comment (JasperCraeghs) [reply mlx/directives/item_matrix_directive.py:143]: That would be better practice indeed. Done in  24abe8320517d0f5706ce037708e6c1f3b43b3c1.

Comment (JasperCraeghs) [reply mlx/directives/item_matrix_directive.py:143]: I believe that those other functions only modify the container variables passed as parameters. They wouldn't be able to create them from scratch. Refactoring them to return a container and merge containers afterwards is not feasible.

**Pull request merged**